python_modules/dagster/dagster/core/host_representation/external_data.py

```diff
@@ -202,27 +202,44 @@ def __new__(cls, error):
 
 
 @whitelist_for_serdes
-class ExternalSensorData(
+class ExternalTargetData(
     namedtuple(
-        "_ExternalSensorData",
-        "name pipeline_name solid_selection mode min_interval description",
+        "_ExternalTargetData",
+        "pipeline_name solid_selection mode",
     )
 ):
     def __new__(
         cls,
-        name,
         pipeline_name,
         solid_selection,
         mode,
+    ):
+        return super(ExternalTargetData, cls).__new__(
+            cls,
+            pipeline_name=check.opt_str_param(pipeline_name, "pipeline_name"),
+            solid_selection=check.opt_nullable_list_param(solid_selection, "solid_selection", str),
+            mode=check.opt_str_param(mode, "mode"),
+        )
+
+
+@whitelist_for_serdes
+class ExternalSensorData(
+    namedtuple(
+        "_ExternalSensorData",
+        "name target_dict min_interval description",
+    )
+):
```

Comment:
careful with these change - i suspect they are breaking

Reply:
I think this is resolved, by having the constructor handle the backcompat.